src/child.c

```diff
@@ -233,6 +233,7 @@ static void child_main (struct child_s *ptr)
 
                 ret = select(maxfd + 1, &rfds, NULL, NULL, NULL);
                 if (ret == -1) {
+                        if (errno == EINTR) continue;
```

Comment:
The change looks reasonable.
But on the cosmetic side, could you put the continue on a line of its own and use braces even around the 1-line block? Thanks!